gcloud-java-compute/src/main/java/com/google/gcloud/compute/ResourceId.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Base class for Google Compute Engine resource identities.
+ */
+public abstract class ResourceId implements Serializable {
+
+  static final String REGEX =
+      "(https?://(www|content).googleapis.com/compute/v1/)?projects/[^/]+/";
```

Comment:
If you are concerned about this (and I think it's a reasonable concern) we could avoid the base url part as a whole. This will make the matching less strict but still sound enough.